lib/adapters/class-searchpress.php

```diff
@@ -132,18 +132,38 @@ public function add_search_suggest_to_indexed_post_data( $data, $sp_post ) {
 
 	/**
 	 * A callback for the sp_search_query_args filter. Adds aggregations to the
-	 * request so that the response will include aggregation buckets.
+	 * request so that the response will include aggregation buckets, and
+	 * filters by any aggregations set.
 	 *
 	 * @param array $es_args The request args to be filtered.
 	 * @return array The filtered request args.
 	 */
 	public function add_aggs_to_es_query( $es_args ): array {
 		// Add our aggregations.
 		foreach ( $this->get_aggregations() as $aggregation ) {
+			// Add aggregations to the request so buckets will be returned with results.
 			$request = $aggregation->request();
 			if ( ! empty( $request ) ) {
 				$es_args['aggs'][ $aggregation->get_query_var() ] = $request;
 			}
+
+			// Add any set aggregations to the query so results are properly filtered.
+			if ( ! empty( $es_args['query']['bool']['must'] ) && is_array( $es_args['query']['bool']['must'] ) ) {
+				$filter = $aggregation->filter();
+				if ( ! empty( $filter ) ) {
+					// If this is a post type query and there is already a post type filter set, remove it.
+					if ( 'post_type' === $aggregation->get_query_var() ) {
+						foreach ( $es_args['query']['bool']['must'] as $key => $must ) {
+							if ( isset( $must['terms']['post_type.raw'] ) ) {
+								unset( $es_args['query']['bool']['must'][ $key ] );
+							}
+						}
+					}
```

Comment:
Thinking aloud here... I don't think we want to unset this, and doing so might be a security issue.

If we leave it, I _think_ the two sets of post types need to overlap, so the results must be within the intersection of this set of post types plus the aggregation's set. If we remove this, it could potentially allow one to query for docs in private post types, since we're not otherwise sanitizing the post type against what we're allowed to search within (as far as I can tell).

... Right?

Reply:
Ah, good call. I had removed the existing post type filter in favor of the new one thinking that it would be more efficient (don't filter by post type twice) but you're right that there's a benefit to keeping the initial list in place and only returning post types that match both. This isn't an issue for other aggregations, I'm only doing this here, and only on this adapter. I'll get this fixed up.

Reply:
Fix applied.